client does not properly send streaming requests

I've run into some issues that suggest a probable bug in `tower-grpc` while working on implementing the [`client_streaming`](https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#client_streaming) interoperability test case. 

This test has the client call a client streaming RPC, sends multiple payloads, and then await a unary response, while the server aggregates the number of bytes received from the client and sends that value in the response. The client then asserts that the server received the correct number of bytes.

Here's the relevant code in the client implementation:

https://github.com/tower-rs/tower-grpc/blob/6a7ce4077200080cbd47b2bb6e209b087316a653/tower-grpc-examples/src/interop/client.rs#L390-L416

When I run the `tower-grpc` test client against `grpc-go`'s interop server, the test fails at the assertion that the bytes aggregated value returned by the server is correct:
```
client_streaming:
  ✔ call must be successful
  ✖ aggregated payload size must be 74922 bytes
  in `response.aggregated_payload_size == 74922`: aggregated_payload_size=0
```

Interestingly, when I run the same test against `grpc-java`'s test server, the server returns an error, and prints the following stack trace to the console:
```
Jan 03, 2018 12:55:27 PM io.grpc.internal.SerializingExecutor run
SEVERE: Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable@3ac811
io.grpc.StatusRuntimeException: INTERNAL: Invalid protobuf byte sequence
	at io.grpc.Status.asRuntimeException(Status.java:517)
	at io.grpc.protobuf.lite.ProtoLiteUtils$2.parse(ProtoLiteUtils.java:168)
	at io.grpc.protobuf.lite.ProtoLiteUtils$2.parse(ProtoLiteUtils.java:82)
	at io.grpc.MethodDescriptor.parseRequest(MethodDescriptor.java:287)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailable(ServerCallImpl.java:251)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable.runInContext(ServerImpl.java:624)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: com.google.protobuf.InvalidProtocolBufferException: Protocol message contained an invalid tag (zero).
	at com.google.protobuf.InvalidProtocolBufferException.invalidTag(InvalidProtocolBufferException.java:105)
	at com.google.protobuf.CodedInputStream$ArrayDecoder.readTag(CodedInputStream.java:609)
	at com.google.protobuf.BoolValue.<init>(BoolValue.java:43)
	at com.google.protobuf.BoolValue.<init>(BoolValue.java:14)
	at com.google.protobuf.BoolValue$1.parsePartialFrom(BoolValue.java:453)
	at com.google.protobuf.BoolValue$1.parsePartialFrom(BoolValue.java:448)
	at com.google.protobuf.CodedInputStream$ArrayDecoder.readMessage(CodedInputStream.java:874)
	at io.grpc.testing.integration.Messages$StreamingInputCallRequest.<init>(Messages.java:4867)
	at io.grpc.testing.integration.Messages$StreamingInputCallRequest.<init>(Messages.java:4809)
	at io.grpc.testing.integration.Messages$StreamingInputCallRequest$1.parsePartialFrom(Messages.java:5666)
	at io.grpc.testing.integration.Messages$StreamingInputCallRequest$1.parsePartialFrom(Messages.java:5661)
	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:91)
	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:49)
	at io.grpc.protobuf.lite.ProtoLiteUtils$2.parseFrom(ProtoLiteUtils.java:173)
	at io.grpc.protobuf.lite.ProtoLiteUtils$2.parse(ProtoLiteUtils.java:165)
	... 9 more
```

This leads me to suspect the request messages are being encoded improperly?

--- a/tower-grpc-interop/README.md
+++ b/tower-grpc-interop/README.md
@@ -8,10 +8,10 @@ Note that currently, only the interop test client is implemented. The `docker-co
 
 - [x] `empty_unary`: implemented in client
 - [ ] `cacheable_unary`: started, requires request context implementation to set cacheable flag
-- [x] `large_unary`: implemented in client, broken due to [#14](https://github.com/tower-rs/tower-grpc/issues/14)
+- [x] `large_unary`: implemented in client
 - [ ] ~`client_compressed_unary`~: requires gRPC compression, NYI
 - [ ] ~`server_compressed_unary`~: requires gRPC compression, NYI
-- [x] `client_streaming`: implemented in client, broken due to [#16](https://github.com/tower-rs/tower-grpc/issues/16)
+- [x] `client_streaming`: implemented in client
 - [ ] ~`client_compressed_streaming`~: requires gRPC compression, NYI
 - [ ] `server_streaming`
 - [ ] ~`server_compressed_streaming`~: requires gRPC compression, NYI

--- a/tower-grpc-interop/src/client.rs
+++ b/tower-grpc-interop/src/client.rs
@@ -31,6 +31,7 @@ use tower_grpc::Request;
 use tower_h2::client::Connection;
 
 use pb::SimpleRequest;
+use pb::StreamingInputCallRequest;
 use pb::client::TestService;
 
 
@@ -278,12 +279,13 @@ impl Testcase {
                 unimplemented!()
             },
             Testcase::client_streaming => {
-                let stream = stream::iter_ok(vec![
-                    util::client_payload(27182),
-                    util::client_payload(8),
-                    util::client_payload(1828),
-                    util::client_payload(45904),
-                ]);
+                let requests = vec![27182, 8, 1828, 45904]
+                    .into_iter()
+                    .map(|len| StreamingInputCallRequest {
+                        payload: Some(util::client_payload(len as usize)),
+                        ..Default::default()
+                    });
+                let stream = stream::iter_ok(requests);
                 core.run(
                     client.streaming_input_call(Request::new(stream))
                         .then(|result| {

--- a/tower-grpc-interop/travis-interop.sh
+++ b/tower-grpc-interop/travis-interop.sh
@@ -24,4 +24,4 @@ trap 'echo ":; killing test server"; kill ${SERVER_PID};' EXIT
 
 # run the interop test client against the server.
 cargo run -p tower-grpc-interop --bin client -- \
-    --test_case=empty_unary,large_unary
+    --test_case=empty_unary,large_unary,client_streaming

